### PR TITLE
SGC-1119 separated system and volume settings

### PR DESCRIPTION
**Pull request by rickardabilia**

### CODE REVIEW

File: android/src/main/kotlin/com/abilia/system_settings_editor/VolumeSettingsHandler.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import android.media.AudioManager
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+
+class VolumeSettingsHandler(context: Context) {
+
+  private val audioManager: AudioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
+
+  internal fun getAlarmVolume(): Double {
+    return audioManager.getStreamVolume(AudioManager.STREAM_ALARM) / getAlarmMaxVolume().toDouble()
```
Comment (bornold) [inline android/src/main/kotlin/com/abilia/system_settings_editor/VolumeSettingsHandler.kt]: Im not sure that it is the ALARM audio channel we are using, and not NOTIFICATION...

Maybe make sure that it affects the "alarms" later

**Review by bornold — DISMISSED** [review]
Looks great! Gonna rebase into this right now

only comment is the comment I left, maybe quick view from @tobiasjunsten before merge?

File: android/src/main/kotlin/com/abilia/system_settings_editor/SystemSettingsEditorPlugin.kt [hunk 2]
```diff
@@ -163,4 +87,19 @@ class SystemSettingsEditorPlugin : FlutterPlugin, MethodCallHandler, ActivityAwa
   override fun onDetachedFromActivity() {
     this.activity = null
   }
+
+  companion object{
+
+    const val METHOD_GET_BRIGHTNESS = "getBrightness"
```
Comment (tobiasjunsten) [inline android/src/main/kotlin/com/abilia/system_settings_editor/SystemSettingsEditorPlugin.kt]: I don't really see the use of these constants. They will only be used in onMethodCall so not really any extra "spelling safety" and a bit harder to map the functionName to the handler when trying to follow the code.

**Review by tobiasjunsten — DISMISSED** [review]
NIce! I like having different handlers for system settings and volume. Just left a comment about having constants for the call method names. If you don't agree you can keep them.

File: android/src/main/kotlin/com/abilia/system_settings_editor/VolumeSettingsHandler.kt (hunk 1 above)
Comment (rickardabilia) [reply android/src/main/kotlin/com/abilia/system_settings_editor/VolumeSettingsHandler.kt]: Seems to be the RING stream, but I'll do some more testing.

File: android/src/main/kotlin/com/abilia/system_settings_editor/SystemSettingsEditorPlugin.kt (hunk 2 above)
Comment (rickardabilia) [reply android/src/main/kotlin/com/abilia/system_settings_editor/SystemSettingsEditorPlugin.kt]: No, you're right. I created them with the intention of using them across the plugin, but it's of course not possible to use them in the dart section. I did some googling to see if there was any way to achieve it, but didn't find any that were not cumbersome.

**Comment by rickardabilia** [comment]
I replaced the ALARM stream with NOTIFICATION. From a user perspective it affects the ring tone (at least on my device). Changing ring tone volume changes mp alarm volume. So I guess using RING stream would have been viable too. But perhaps it differs on different sets.

**Review by bornold — APPROVED** [review]
⬆️🔉⬇️ 
    💯

**Pull request merged**